src/ghga_datasteward_kit/s3_upload/utils.py

```diff
@@ -34,11 +34,11 @@
 class HttpxClientConfig:
     """Helper for user configurable httpx request parameters."""
 
-    num_retries: int = 5
-    timeout: int = 60
+    num_retries: int
```

Comment:
When you disable the retry, the `HttpxClientConfig.configure()` is not called, and there is no default value for `num_retries` so it remains none.

Reply:
Yeah, fixed the tests just now, waiting for CI to finish